R/utils.R

```diff
@@ -324,6 +324,145 @@ date_to_numeric_modulus <- function(dt, date_column, timestep) {
   return(dt[])
 }
 
+#' Set caching location for Stan models
+#'
+#' This function allows the user to set a cache location for
+#' Stan models rather than a temp directory. This can reduce the
+#' need for model compilation on every new model run across sessions.
+#'
+#' @param path A valid filepath representing the desired cache location
+#'
+#' @return The string of the filepath set
+#' @examples
+#' # Set to local directory
+#' dir.create("test")
+#' enw_set_cache("test")
+#' enw_get_cache()
+#' # Use  the package cache in R >= 4.0
+#' if(R.version.string >= 4.0) {
+#'   enw_set_cache(tools::R_user_dir(package = "epinowcast", "cache"))
+#' }
+#' @importFrom cli cli_abort cli_alert
+#' @export
+
+enw_set_cache <- function(path = NULL) {
+
+  if (is.null(path)) {
+    cli::cli_abort("`path` must be a valid file path.")
```

Comment:
this check kind of implies we have checked if it is a valid file path which we haven't. Do you think we should or do anything else here?

Reply:
the `cmdstanr::write_stan_file` function is pretty aggressive in creating a directory if it does exist. I could add another message that triggers on a `dir.exists` which could do a "directory does not exist, it will be created" note?

Reply:
i think we can use the base R default error for missing arguments work here, rather than setting this to null and checking for null.

then a basic is-it-a-string check before we toss to cmdstanr seems sufficient.

we *could* get into writing a sophisticated directory structure checking / sanitization bit here, but that seems like some other package's job.